inference/wasm/tests/engine/translations-engine.worker.mjs

```diff
@@ -423,7 +423,30 @@ function cleanText(sourceText) {
   const whitespaceAfter = result[3];
   let cleanedSourceText = result[2];
 
-  cleanedSourceText = cleanedSourceText.replace(/\u00AD/g, "");
+  // Remove all soft hyphens from the text.
```

Comment:
Suggestion: I'm a little hesitant to micro-optimize non-production code, but since this is supposed to be close to what we ship in Gecko, we should only do this for locales where it matters. So here we would check for a locale before applying these fixes. I may leave these same comments on the Gecko side when I review it.

Reply:
(See reply on other comment)